exist-versioning-release.md

```diff
@@ -93,40 +93,121 @@ That version number is also copied into the `META-INF/MANIFEST.MF` file of any J
+----------------------------------
+sec   rsa4096/12345678 2013-04-26 [SC]
+```
+
+This gives you the keystore you are using (here: `/Users/username/.gnupg/pubring.kbx`) and your key-id (here: `12345678` the part after rsa4096/). 
+
+**Important Note** 
+GnuPG is using an [optimized keystore since version 2.1](https://www.gnupg.org/documentation/manuals/gnupg/GPG-Configuration.html) which contains public and secret keys. Former GnuPG version had seperated keystore for public (~/.gnupg/pubring.gpg) and private keys (~/.gnupg/secring.gpg). 
+
+###### Maven Keystore values prior GnuPG 2.1
+
+* existdb.release.public-keyfile=~/.gnupg/pubring.gpg
+* existdb.release.private-keyfile=~/.gnupg/secring.gpg
+
+###### Maven Keystore values for GnuPG 2.1 or newer
+
+* existdb.release.public-keyfile=~/.gnupg/pubring.kbx
+* existdb.release.private-keyfile=~/.gnupg/pubring.kbx
+                   
+To test gpg simply create a file `test.txt` and run`gpg -ab test.txt`to sign it. you should see a new file `test.txt.asc` which means GnuPG signing worked fine. 
+
+##### Java KeyStore 
+* To create an eXist-db release a Java KeyStore with key is needed to sign the IzPack Installer. In case you don't have a Java KeyStore you can create one with: 
+
+```bash
+keytool -genkey \
+  -alias some-alias \
+  -keystore keystore.p12 \
+  -storetype PKCS12 \
+  -keyalg RSA \
+  -storepass some-pass \
+  -validity 7300 \
+  -keysize 4096
+```
+
+To verify if your Java KeyStore works fine run 
+
+```bash
+keytool -list -v -keystore /path/to/keystore.p12
+```
 
-1.  Modify `$EXIST_HOME/build.properties` to read:
+##### Apple Developer Certificate
 
+A valid Apple Developer Certificate to sign Mac DMG is required. If you don't have one you will need to join the [Apple Developer Program](https://developer.apple.com/programs/enroll/) and to create a certificate via XCode. 
+
+##### Github
+
+A Github account and username / password or Github Personal (access tokens)[https://github.com/settings/tokens] with permission to publish Github releases to the eXist-db org. 
+
+##### Sonatype OSS
+
+You will need a Sonatype OSS acount which has been authorized for the eXist-db organisation at Sonatype OSS staging for Maven Central - https://oss.sonatype.org/
+
+
+
+##### Docker 
+
+A DockerHub user with permissions to publish to https://cloud.docker.com/orgs/existdb/ is required. 
+
+
+#### Prepare Release Notes 
+
+* There are various possibilities to create the release notes.
+
+> This section is still work in progress and manual tasks are needed to generate the release notes
+
+The release notes start with an introduction summarizing the most important changes for end-users and developers since the last version was released.
+
+Three sections follow
+
+- a list of all changes and features with links to their PRs
+- a list of all fixes with links the their PRs
+- a list of all dependencies that got updated along with the new version that is now used in existdb
```

Comment:
Maybe a note to say that this may need to be deduplicated as the same dependency may have been updated more than once between releases. It is important to list both the version from the last release, and the newest version that goes into this release.